doc/modules/feature_extraction.rst

```diff
@@ -101,6 +101,81 @@ memory the ``DictVectorizer`` class uses a ``scipy.sparse`` matrix by
+    class is experimental and the API is subject to change.
+
+Many datasets contain features of different types, say text, floats, and dates,
+where each type of feature requires separate preprocessing or feature extraction steps.
+Often it is easiest to preprocess data before applying scikit-learn methods,
+for example using `pandas <http://pandas.pydata.org/>`__.
+The :class:`~sklearn.experimental.ColumnTransformer` is a convenient way to perform heterogeneous
+preprocessing on data columns within a scikit-learn pipeline; e.g. when one wants
+to adjust preprocessing parameters within a grid search or to
+avoid leaking statistics from test data to training data.
+The :class:`~sklearn.experimental.ColumnTransformer` works on arrays, sparse matrices,
+and pandas DataFrames.
+
+To each column, a different transformation can be applied, such as
+preprocessing or a specific feature extraction method::
+
+  >>> import pandas as pd
+  >>> X = pd.DataFrame(
+  ...     {'city': ['London', 'London', 'Paris', 'New York'],
+  ...      'title': ["His Last Bow", "How Watson Learned the Trick",
+  ...                "A Moveable Feast", "The Great Gatsby"]})
+
+For this data, we might want to encode ``'city'`` column as a categorical variable,
+but apply a :class:`feature_extraction.text.CountVectorizer <sklearn.feature_extraction.text.CountVectorizer>`
+to the ``'title'`` column.
+As we might use multiple feature extraction methods on the same column, we give each
+transformer a unique name, say ``'city_category'`` and ``'title_bow'``::
+
+  >>> from sklearn.experimental import ColumnTransformer
+  >>> from sklearn.feature_extraction.text import CountVectorizer
+  >>> column_trans = ColumnTransformer(
+  ...     [('city_category', CountVectorizer(analyzer=lambda x: [x]), 'city'),
+  ...      ('title_bow', CountVectorizer(), 'title')])
+
+  >>> column_trans.fit(X) # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+  ColumnTransformer(n_jobs=1, transformer_weights=None,
+      transformers=...)
+
+  >>> column_trans.get_feature_names() == [
+  ...    'city_category__London', 'city_category__New York', 'city_category__Paris',
+  ...    'title_bow__bow', 'title_bow__feast', 'title_bow__gatsby',
+  ...    'title_bow__great', 'title_bow__his', 'title_bow__how', 'title_bow__last',
+  ...    'title_bow__learned', 'title_bow__moveable', 'title_bow__the',
+  ...    'title_bow__trick', 'title_bow__watson']
+  True
+
+  >>> column_trans.transform(X).toarray() # doctest: +NORMALIZE_WHITESPACE +ELLIPSIS
+  array([[1, 0, 0, 1, 0, 0, 0, 1, 0, 1, 0, 0, 0, 0, 0],
+         [1, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 0, 1, 1, 1],
+         [0, 0, 1, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0],
+         [0, 1, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0]]...)
+
+In the above example, the :class:`~sklearn.feature_extraction.text.CountVectorizer`
+expects a 1D array as input and therefore the columns were specified as a scalar (``'city'``).
```

Comment:
Maybe rather than scalar, call this a string, and below, a list of strings. I realise this might not be as precise and does not generalise to integer columns, by I think it is easier to grasp the distinction that way